rclcpp/src/rclcpp/clock.cpp

```diff
@@ -115,66 +99,51 @@ Clock::get_clock_type()
   return rcl_clock_.type;
 }
 
-rclcpp::JumpHandler::SharedPtr
-Clock::create_jump_callback(
-  std::function<void()> pre_callback,
-  std::function<void(const TimeJump &)> post_callback,
-  JumpThreshold & threshold)
+void
+Clock::on_time_jump(
+  const struct rcl_time_jump_t * time_jump,
+  bool before_jump,
+  void * user_data)
 {
-  // JumpHandler jump_callback;
-  auto jump_callback =
-    std::make_shared<rclcpp::JumpHandler>(pre_callback, post_callback, threshold);
-  {
-    std::lock_guard<std::mutex> guard(callback_list_mutex_);
-    active_jump_handlers_.push_back(jump_callback);
+  rclcpp::JumpHandler * handler = static_cast<rclcpp::JumpHandler *>(user_data);
+  if (before_jump && handler->pre_callback) {
+    handler->pre_callback();
+  } else if (handler->post_callback) {
```

Comment:
The conditions in the `if` and `else if` look weird. If `before_jump` is `true` but the `handler` only has a post callback it is being called here?

Reply:
Added test and fixed in af09c599b61f5e73baa204d78705a87cfed32779